### PR TITLE
ZipArchive updates

### PR DESCRIPTION
**Pull request by mairaw**

### CODE REVIEW

File: snippets/visualbasic/VS_Snippets_CLR_System/system.io.compression.ziparchive/vb/program1.vb [hunk 1]
```diff
@@ -6,16 +6,38 @@ Module Module1
...
+        Dim extractPath As String = Console.ReadLine()
+
+        ' Normalizes the path.
+        extractPath = Path.GetFullPath(extractPath)
+
+        ' Ensures that the last character on the extraction path
+        ' is the directory separator char. 
+        ' Without this, a malicious zip file could try to traverse outside of the expected
+        ' extraction path.
+        If (extractPath(extractPath.Length - 1) <> Path.DirectorySeparatorChar) Then
```
Comment (rpetrusha) [inline snippets/visualbasic/VS_Snippets_CLR_System/system.io.compression.ziparchive/vb/program1.vb]: The enclosing parens aren't needed here. This should be just:
`If extractPath(extractPath.Length - 1 <> Path.DirectorySeparatorChar Then`

File: snippets/visualbasic/VS_Snippets_CLR_System/system.io.compression.ziparchive/vb/program2.vb [hunk 2]
```diff
@@ -6,12 +6,34 @@ Module Module1
...
+        Dim extractPath As String = Console.ReadLine()
+
+        ' Normalizes the path.
+        extractPath = Path.GetFullPath(extractPath)
+
+        ' Ensures that the last character on the extraction path
+        ' is the directory separator char. 
+        ' Without this, a malicious zip file could try to traverse outside of the expected
+        ' extraction path.
+        If (extractPath(extractPath.Length - 1) <> Path.DirectorySeparatorChar) Then
```
Comment (rpetrusha) [inline snippets/visualbasic/VS_Snippets_CLR_System/system.io.compression.ziparchive/vb/program2.vb]: Here again, no need for the enclosing parens

File: snippets/csharp/VS_Snippets_CLR_System/system.io.compression.ziparchive/cs/program1.cs [hunk 3]
```diff
@@ -10,19 +10,37 @@ class Program
...
+            string extractPath = Console.ReadLine();
+            
+            // Normalizes the path.
+            extractPath = Path.GetFullPath(extractPath);
+
+            // Ensures that the last character on the extraction path
+            // is the directory separator char. 
+            // Without this, a malicious zip file could try to traverse outside of the expected
+            // extraction path.
+            if (extractPath[extractPath.Length - 1] != Path.DirectorySeparatorChar)
```
Comment (BillWagner) [inline snippets/csharp/VS_Snippets_CLR_System/system.io.compression.ziparchive/cs/program1.cs]: The `string.EndsWith` method might be easier to use here.

File: snippets/csharp/VS_Snippets_CLR_System/system.io.compression.ziparchive/cs/program2.cs [hunk 4]
```diff
@@ -10,19 +10,37 @@ class Program
...
+            string extractPath = Console.ReadLine();
+            
+            // Normalizes the path.
+            extractPath = Path.GetFullPath(extractPath);
+
+            // Ensures that the last character on the extraction path
+            // is the directory separator char. 
+            // Without this, a malicious zip file could try to traverse outside of the expected
+            // extraction path.
+            if (extractPath[extractPath.Length - 1] != Path.DirectorySeparatorChar)
```
Comment (BillWagner) [inline snippets/csharp/VS_Snippets_CLR_System/system.io.compression.ziparchive/cs/program2.cs]: same comment on `EndsWith`

That would also apply to the VB code in this PR

**Comment by mairaw** [comment]
Thanks guys for the feedback!

**Pull request merged**